commit b17b7101e21060501683ae1ac6689f8c328670ce
Author: Marc Zyngier <marc.zyngier@arm.com>
Date:   Fri Aug 18 10:53:45 2017 +0100

    genirq: Restore trigger settings in irq_modify_status()
    
    commit e8f241893dfbbebe2813c01eac54f263e6a5e59c upstream.
    
    irq_modify_status starts by clearing the trigger settings from
    irq_data before applying the new settings, but doesn't restore them,
    leaving them to IRQ_TYPE_NONE.
    
    That's pretty confusing to the potential request_irq() that could
    follow. Instead, snapshot the settings before clearing them, and restore
    them if the irq_modify_status() invocation was not changing the trigger.
    
    Fixes: 1e2a7d78499e ("irqdomain: Don't set type when mapping an IRQ")
    Reported-and-tested-by: jeffy <jeffy.chen@rock-chips.com>
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Cc: Jon Hunter <jonathanh@nvidia.com>
    Link: http://lkml.kernel.org/r/20170818095345.12378-1-marc.zyngier@arm.com
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1653_b17b7101e21060501683ae1ac6689f8c328670ce.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1653_b17b7101e21060501683ae1ac6689f8c328670ce.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
